administrator/components/com_associations/views/association/tmpl/edit.php

```diff
@@ -121,27 +121,37 @@
 				var reference     = document.getElementById('reference-association');
 				var referenceId   = reference.getAttribute('data-id');
 				var languageCode  = reference.getAttribute('data-language').replace(/-/, '_');
+				var title         = $(reference).contents().find('#jform_title').val();
+				target = $(this).contents();
 
 				// - For modal association selectors.
-				$(this).contents().find('#jform_associations_' + languageCode + '_id').val(referenceId);
-				$(this).contents().find('#jform_associations_' + languageCode + '_name').val($(reference).contents().find('#jform_title').val());
+				target.find('#jform_associations_' + languageCode + '_id').val(referenceId);
+				target.find('#jform_associations_' + languageCode + '_name').val(title);
+
+				// - For chosen association selectors (menus).
+				target.find('#jform_associations_' + languageCode + '_chzn').remove();
+				chznField = target.find('#jform_associations_' + languageCode);
+				chznField.append('<option value=\"'+ referenceId + '\">' + title + '</option>');
+				chznField.val(referenceId).change().chosen();
 
-				target = $(this).contents();
 
 				$('#jform_itemlanguage option').each(function()
 				{
 					parse = $(this).val().split('|');
 
 					if (typeof parse[1] !== 'undefined' && parse[1] !== '0')
 					{
+						// - For modal association selectors.
 	 					langAssociation = parse[0].replace(/-/,'_');
 	 					target.find('#jform_associations_' + langAssociation + '_id').val(parse[1]);
+
+	 					// - For chosen association selectors (menus).
+						target.find('#jform_associations_' + langAssociation + '_chzn').remove();
+						chznField = target.find('#jform_associations_' + langAssociation);
+						chznField.append('<option value=\"'+ parse[1] + '\"></option>');
+						chznField.val(parse[1]).change().chosen();
 	 				}
```

Comment:
Doing si!ikar operatons before and in the cycle. Can't we do all inside the the cycle